src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -147,6 +151,18 @@ private void updateActionForDeployment(Map<String, Object> newConfig, Deployment
             return;
         }
 
+        // As long as the timeout is not 0, we will try to run the connectivity validation
+        Integer timeout = deployment.getDeploymentDocumentObj().getConfigurationValidationPolicy().timeoutInSeconds();
```

Comment:
check for null in getConfigurationValidationPolicy and timeoutInseconds use `Coerce.toInt`